src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -4103,21 +4103,29 @@ class SymbolTableVisitor : public CommonVisitor<SymbolTableVisitor> {
             for (size_t i=0; i<x.n_body; i++) {
                 visit_stmt(*x.m_body[i]);
             }
-
             LCOMPILERS_ASSERT(module_sym != nullptr);
             module_sym->m_dependencies = current_module_dependencies.p;
             module_sym->n_dependencies = current_module_dependencies.size();
             if (!overload_defs.empty()) {
                 create_GenericProcedure(x.base.base.loc);
             }
         } else {
-            ASR::Module_t* module_sym = 
-                ASR::down_cast<ASR::Module_t>(parent_scope->resolve_symbol(module_name));
+            module_sym = ASR::down_cast<ASR::Module_t>(parent_scope->resolve_symbol(module_name));
             LCOMPILERS_ASSERT(module_sym != nullptr);
             current_scope = module_sym->m_symtab;
             for (size_t i=0; i<x.n_body; i++) {
                 visit_stmt(*x.m_body[i]);
             }
+            for (size_t i = 0; i < module_sym->n_dependencies; i++) {
+                if (module_sym->m_dependencies[i]) {
+                    current_module_dependencies.push_back(al, module_sym->m_dependencies[i]);
+                }
+            }
+            module_sym->m_dependencies = current_module_dependencies.p;
+            module_sym->n_dependencies = current_module_dependencies.size();
+            if (!overload_defs.empty()) {
+                create_GenericProcedure(x.base.base.loc);
+            }
```

Comment:
This portion seems to be the same as that of the `if` part. So, let's move this out of the `if-else`?

Reply:
Yes, we should be able to do that. Will do it in the next commit.